compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -105,34 +160,41 @@ class ExtractDependencies extends Phase {
    */
   def recordDependency(currentSourceFile: File, dep: Symbol, context: DependencyContext)
       (implicit ctx: Context) = {
-    val depFile = dep.associatedFile
-    if (depFile != null) {
-      if (depFile.path.endsWith(".class")) {
-        /** Transform `List(java, lang, String.class)` into `java.lang.String` */
-        def className(classSegments: List[String]) =
-          classSegments.mkString(".").stripSuffix(".class")
-        def binaryDependency(file: File, className: String) =
-          ctx.sbtCallback.binaryDependency(file, className, extractedName(currentClass), currentSourceFile, context)
-
-        depFile match {
-          case ze: ZipArchive#Entry =>
-            for (zip <- ze.underlyingSource; zipFile <- Option(zip.file)) {
-              val classSegments = Path(ze.path).segments
-              binaryDependency(zipFile, className(classSegments))
-            }
-          case pf: PlainFile =>
-            val packages = dep.ownersIterator
-              .filter(x => x.is(PackageClass) && !x.isEffectiveRoot).length
-            // We can recover the fully qualified name of a classfile from
-            // its path
-            val classSegments = pf.givenPath.segments.takeRight(packages + 1)
-            binaryDependency(pf.file, className(classSegments))
-          case _ =>
-            ctx.warning(s"sbt-deps: Ignoring dependency $depFile of class ${depFile.getClass}")
-        }
-      } else if (depFile.file != currentSourceFile) {
-        ctx.sbtCallback.classDependency(extractedName(dep.enclosingClass), extractedName(currentClass), context)
+    val onSource = dep.sourceFile
```

Comment:
Why was this changed? This is also very likely to do the wrong thing. If `dep` was loaded from TASTY, `dep.sourceFile` will return the source filename stored in the SoureFile attribute of the TASTY section. But here we're really interested in recording the dependency on the classfile where `dep` comes from, because if this classfile changes we'll have to recompile the current unit.